File: src/main/java/dev/zprestige/fire/module/misc/freelook/CameraSetupListener.java
package dev.zprestige.fire.module.misc.freelook;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.CameraSetupEvent;

public class CameraSetupListener extends EventListener<CameraSetupEvent, Freelook> {

    public CameraSetupListener(final Freelook freelook){
        super(CameraSetupEvent.class, freelook);
    }

    @Override
    public void invoke(final Object object){
        final CameraSetupEvent event = (CameraSetupEvent) object;
        event.setYaw(event.getYaw() + module.yaw);
        event.setPitch(event.getPitch() + module.pitch);
    }
}

File: src/main/java/dev/zprestige/fire/module/misc/freelook/CameraSetupListener.kt
package dev.zprestige.fire.module.misc.freelook

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.CameraSetupEvent

class CameraSetupListener(freelook: Freelook) : EventListener<CameraSetupEvent, Freelook>(CameraSetupEvent::class.java,
    freelook
) {

    override fun invoke(e: Any) {
        val event = e as CameraSetupEvent
        event.yaw = event.yaw + module.yaw
        event.pitch = event.pitch + module.pitch
    }
}